frame/nomination-pools/src/lib.rs

```diff
@@ -1601,13 +1621,69 @@ pub mod pallet {
 			Ok(())
 		}
 
+		/// Bond pending rewards of `member_account` into the pool they already belong.
+		///
+		/// Note: `member_account` must pass `RewardClaim::Permissionless` to `set_reward_claim`,
+		/// making this call permissionless.
+		#[pallet::call_index(2)]
+		#[pallet::weight(
+			T::WeightInfo::bond_extra_transfer()
+			.max(T::WeightInfo::bond_extra_reward())
+		)]
+		pub fn bond_extra_pending_rewards_other(
+			origin: OriginFor<T>,
+			member_account: AccountIdLookupOf<T>,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+			let member_account = T::Lookup::lookup(member_account)?;
+			let (mut member, mut bonded_pool, mut reward_pool) =
+				Self::get_member_with_pools(&member_account)?;
+
+			ensure!(
+				RewardClaimPermission::<T>::get(&member_account) == RewardClaim::Permissionless,
+				Error::<T>::DoesNotHavePermission
+			);
+
+			reward_pool.update_records(bonded_pool.id, bonded_pool.points)?;
+			ensure!(!member.active_points().is_zero(), Error::<T>::FullyUnbonding);
+
+			let current_reward_counter =
+				reward_pool.current_reward_counter(bonded_pool.id, bonded_pool.points)?;
+			let pending_rewards = member.pending_rewards(current_reward_counter)?;
+
+			if !pending_rewards.is_zero() {
+				member.last_recorded_reward_counter = current_reward_counter;
+				reward_pool.register_claimed_reward(pending_rewards);
+			}
+
+			let points_issued = bonded_pool.try_bond_funds(
+				&bonded_pool.reward_account(),
+				pending_rewards,
+				BondType::Later,
+			)?;
+
+			bonded_pool.ok_to_be_open()?;
+			member.points = member.points.saturating_add(points_issued);
+
+			Self::deposit_event(Event::<T>::Bonded {
+				member: member_account.clone(),
+				pool_id: member.pool_id,
+				bonded: pending_rewards,
+				joined: false,
+			});
+
+			Self::put_member_with_pools(&member_account, member, bonded_pool, reward_pool);
+
+			Ok(())
+		}
+
 		/// A bonded member can use this to claim their payout based on the rewards that the pool
 		/// has accumulated since their last claimed payout (OR since joining if this is there first
 		/// time claiming rewards). The payout will be transferred to the member's account.
 		///
 		/// The member will earn rewards pro rata based on the members stake vs the sum of the
 		/// members in the pools stake. Rewards do not "expire".
-		#[pallet::call_index(2)]
+		#[pallet::call_index(3)]
```

Comment:
For nom pool its probably not that critical, but the joke of having explicit call indices is that they are not meant to change.  
So you can either: Put your new Calls at the bottom of the pallet or give them the correct call indices.  
`bond_extra_pending_rewards_other` would get 14.

Reply:
thank you for explaining this @ggwpez